src/modules/map/components/cesium/CesiumGPXLayer.vue

```diff
@@ -0,0 +1,154 @@
+            getViewer().dataSources.add(dataSource)
+            isPresentOnMap.value = true
+        })
+        .then(updateStyle)
+}
+
+function removeLayer() {
+    log.debug('Remove GPX layer')
+    if (gpxDataSource) {
+        getViewer().dataSources.remove(gpxDataSource)
+        gpxDataSource = null
+        getViewer().scene.requestRender() // Request a render after removing the DataSource
+    }
+    isPresentOnMap.value = false
+}
+
+// Function to create a red circle image using a canvas
+function createRedCircleImage(radius, opacity = 1) {
+    // Create a new canvas element
+    const canvas = document.createElement('canvas')
+    const context = canvas.getContext('2d')
+
+    // Set the canvas sizes
+    canvas.width = radius * 2
+    canvas.height = radius * 2
+
+    // Draw a red circle on the canvas
+    context.beginPath()
+    context.arc(radius, radius, radius, 0, 2 * Math.PI, false)
+
+    const color = `rgba(255, 0, 0, ${opacity})`
+    context.fillStyle = color
+    context.fill()
+
+    // Return the data URL of the canvas drawing
+    return canvas.toDataURL()
+}
+
+function updateStyle() {
+    // Create a red circle image with a radius of 8 pixels
+    const radius = 8
+    const billboardSize = radius * 2
+    const redCircleImage = createRedCircleImage(radius, opacity.value)
+
+    const redCircleBillboard = new BillboardGraphics({
+        image: redCircleImage,
+        width: billboardSize,
+        height: billboardSize,
+        eyeOffset: new Cartesian3(0, 0, -100), // Make sure the billboard is always seen
+        heightReference: HeightReference.CLAMP_TO_TERRAIN, // Make the billboard always appear on top of the terrain
+    })
+    const redColorMaterial = new ColorMaterialProperty(Color.RED.withAlpha(opacity.value))
+
+    const entities = gpxDataSource.entities.values
+
+    entities.forEach((entity) => {
+        if (opacity.value === 0) {
+            entity.show = false
+            return
+        }
+        // Hide the billboard for billboard on the lines by checking if there is a description
+        // Imported GPX files from web-mapviewer have a description for the waypoints
+        // This might be not working for generic GPX files
+        if (cesiumDefined(entity.billboard)) {
+            if (!entity.description) {
+                entity.show = false
+            } else {
+                entity.show = true
+                entity.billboard = redCircleBillboard
+            }
+        }
+
+        if (cesiumDefined(entity.polyline)) {
+            entity.show = true
+            entity.polyline.material = redColorMaterial
+            entity.polyline.width = 1.5
+        }
+
+        if (cesiumDefined(entity.polygon)) {
+            entity.show = true
+            entity.polygon.material = redColorMaterial
+            entity.polygon.outline = true
+            entity.polygon.outlineColor = Color.BLACK
+        }
+    })
+    getViewer().scene.requestRender()
```

Comment:
I don't know why, but if I change the opacity of a GPX layer it disappear from the map, and then comes back again if I change the camera position (if I force another render?)

You might want to test that a bit further, if you want a GPX to test, [here's some available file](https://www.crossingswitzerland.com/map)

A wrap with a `nextTick` maybe?

Reply:
ah yes, with a bigger GPX file, I can see the issue you mentioned. I will investigate it.

Reply:
hi @pakb 
Unfortunately, `nextTick` doesn't work. If I forced render it to quickly, it's not working also. I came up with [hacky/dirty solution](https://github.com/geoadmin/web-mapviewer/pull/873/commits/403fffae881c163f9f25ec831cb5d0de52bed419) to do force rendering after some time (1500ms). It's working although there is a delay, but even in [sandcastle](https://sandcastle.cesium.com/#c=fVZtb9s2EP4rRL5UBmJKUeLUdZNsRdZ1G9K1aJpi2Dy0lEhZRGmSIKkk7pD/viMpWVLsJEAA6+6543OvZKmkdeiWsztm0DmS7A5dMsubNf4SZMnyoAzfl0o6wiUzy4PJ66VMU3RNJC2JdYKhglheIstco5cyKK8Uoejdx78QJY4s5S0xaKXvb4yAQ5YHtXPaLtL0dJ5lpzN6OmVsPp+ezOZH0/m8yqezjB7nOauqYlbhigtmG2twqdZps6JplpNZ8fLrbP5qdpyXLD+en5y8rObFUfYqP2bH5Wmek+JVhuHAn6g8L42ylsuVvePuBzMCaOdZfvI1yY6mWT71vyceuzx4jYD5J6YFKRkCdI1czRApXUMEuvl0hZxCG9WYEJin5YNtQ/sF4rwGHViO0vhuqMIC0pLEREwwOJdJ1cjScSVRspQoZCtCl3KC/vOiWBvcaywmlCb9ty+HBwL3v5Vae5KetucoyAYK1jv5AfrPar/tpWHEQbSSIvAfXDilRAHhVcogpUnJ3QaV0AdGieAUVB3kHFFVNmsmHS6Do7eC+S/oH8pvY8+gDo1LQaz9k6xZ6IbYYNNWCWUImejcFYpuMNGaSXpZc0GTFrdL3VPuaApSsC3JVnjlZc8xjUYt16ERduze+QkAXKD8ISoXLdkuriHNof3zXLnUjXvE9Xcve45rNBpzDUbYbXTMqyFyxVqGI8SaSwBkexTkHhRHuwrrmPYmONujvCWiYVu7Z3IR4INcXDOHGr2bC8RufaIFh2NlbODRgdCfbz3iqgX06ThE/Ty18zNKK7DUxFj2KwyiS3aj6MgFejeaPi6VqhARAsHZ3HFmge521vxQIdvOrvfQTxnu8PEQi2Gg3pKyHgx/QGy2lBHiFUraFUJZBYuXthisldgIEEwGaOTnEhZ52RgDsEsllJ/JRxZQX8cMJwJWKQDwirkvnlDSu0Hd3vqjEZxIWF0MS3WXTHpIbLn45xta6U1IQvAZ9kdZ+85D3NkucY95woLsOA52ZRCNyAzjwYbRwyeVK8OYfFpdQJgj7Q6xyWj7vwBa7ZC/iFdBWPwUiAKNbT+EenYt0ybEd7XPhw8s5qQg5fduLceSvLCoq0Vv+1S19iTpfav6aJRmxm2SLqF9cR56Wmn6dDcVXIhCEUNjO/Um6DEgNk3fVY8U2GfpjdA16eaq55KmWzYPUfowWANvtBZ+8mFEINhBYUbzSbnVxJV1mHwfL4q/+lXoXT50jxPnc7gKV9fgKovXdWntc6vVgi2LqxWQ/T71l0AKkrBSvYZLWD6/fX5/BepvMTw8vsyGE6qV5X7YF4gUVonGscEgOaUX6CjT9wOZYJXbEWpYfhDYjtz32MqoRtIFMquCJCf5Ier+MzwfTm2hDGVmagjljV2g2chRKOYC3dV8S/Bhf2zhuhxGuCZmxeXU8FXthm6fsA91+8en93x7Wf079HfHqat9pNnQ1Tf4ua1dzQgdXTVQFx/qweHBWajjRefuZ77WysCFY0SCcerYGibdMZsWTfmdQRdEQw89S4emZ/CAQZye73kJo/CQAU3VCHHNf0AAF2cp4HdM/cMPyvbh1j9ANx5WH11cRSHG+CyFz/2W22fRSP8/), it still takes time to render. 

if I set `requestRenderMode` to false, obviusly it works, but I am sure we don't want to set it to false.

What do you think? Perhaps you know something else to wait until entities are updated?